src/adapters/vfs/system.js

```diff
@@ -36,20 +36,36 @@ const chokidar = require('chokidar');
 /*
  * Creates an object readable by client
  */
-const createFileIter = (core, realRoot, file) => {
+const createFileIter = (core, realRoot, file, options = {}) => {
   const filename = path.basename(file);
   const realPath = path.join(realRoot, filename);
   const {mime} = core.make('osjs/vfs');
 
-  const createStat = stat => ({
-    isDirectory: stat.isDirectory(),
-    isFile: stat.isFile(),
-    mime: stat.isFile() ? mime(realPath) : null,
-    size: stat.size,
-    path: file,
-    filename,
-    stat
-  });
+  const createStat = async stat => {
```

Comment:
This method of figuring out the total size of a directory seems very inefficient, especially when there's a lot of files.

Maybe it would be better looking into spawning a system process to do this in a native way ? :thinking:

Reply:
As I searched, `du` can be used liked this:
```js
const child = spawn('sh', ['-c', 'du -s ${realPath} | cut -f1']);
```
But the problem is that `du` seems to have considerable delay, especially for large directories.

Reply:
> especially for large directories.

But I would assume it's faster than doing it with node, right?

Reply:
Also I'm not 100% sure, but I *assume* that this wouldn't be cross-compatible with users hoping to run OS.js locally on Windows. IMHO native system processes shouldn't be included without a fallback for Windows

Reply:
There's a du npm package that handles Windows etc.

Reply:
In new commit du package is used for totalSize. @andersevenrud

Reply:
I am thinking about our own vfs system adapter, which is a cloud object storage.
Since `totalSize` is not provided in APIs, I have to iterate through files which as you said, gets lots of time. Is there any other solution?

Reply:
Sorry, I can't think of a good solution to that, especially if there's pagination involved :/